src/libs/actions/connections/index.ts

```diff
@@ -0,0 +1,127 @@
+import Onyx from 'react-native-onyx';
+import type {OnyxUpdate} from 'react-native-onyx';
+import * as API from '@libs/API';
+import type {RemovePolicyConnectionParams, UpdatePolicyConnectionConfigParams} from '@libs/API/parameters';
+import {WRITE_COMMANDS} from '@libs/API/types';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {ConnectionName, Connections, PolicyConnectionName} from '@src/types/onyx/Policy';
+
+function removePolicyConnection(policyID: string, connectionName: PolicyConnectionName) {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    [connectionName]: null,
+                },
+            },
+        },
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.POLICY_CONNECTION_SYNC_PROGRESS}${policyID}`,
+            value: null,
+        },
+    ];
+    const failureData: OnyxUpdate[] = [
+        // {
+        //     onyxMethod: Onyx.METHOD.MERGE,
+        //     key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+        //     value: {
+        //         errorFields: {
+        //             avatar: ErrorUtils.getMicroSecondOnyxError('avatarWithImagePicker.deleteWorkspaceError'),
+        //         },
+        //     },
+        // },
+    ];
+    const parameters: RemovePolicyConnectionParams = {
+        policyID,
+        connectionName,
+    };
+    API.write(WRITE_COMMANDS.REMOVE_POLICY_CONNECTION, parameters, {optimisticData, failureData});
+}
+function updatePolicyConnectionConfig<TConnectionName extends ConnectionName, TSettingName extends keyof Connections[TConnectionName]['config']>(
```

Comment:
Did we do too much TypeScript here? 😄

Reply:
haha, I though it could cause problems when using the function `updatePolicyConnectionConfig`, but it didn't and works pretty well on matching the `settingName` with the possible `settingValues` for a specific connection name.